srml/contract/src/wasm/runtime.rs

```diff
@@ -228,14 +248,15 @@ define_env!(Env, <E: Ext>,
 	//   where the value to set is placed. If `value_non_null` is set to 0, then this parameter is ignored.
 	// - value_len: the length of the value. If `value_non_null` is set to 0, then this parameter is ignored.
 	ext_set_storage(ctx, key_ptr: u32, value_non_null: u32, value_ptr: u32, value_len: u32) => {
-		let key = read_sandbox_memory(ctx, key_ptr, 32)?;
+		let mut key = [0; 32];
```

Comment:
I would at least make it `key: StorageKey` to make sure we are informed when the length of the key changes.